pkg/factory/provider.go

```diff
@@ -65,11 +66,25 @@ type testProvider struct {
 	instance2node map[string]string
 }
 
-func GetTestProvider() *testProvider {
+func GetTestProvider(params map[string]string) (*testProvider, error) {
 	p := &testProvider{}
-	p.tree, p.instance2node = translate.GetTreeTestSet(false)
 
-	return p
+	var modelPath string
+	if len(params) != 0 {
+		modelPath = params[common.KeyModelPath]
+	}
+
+	if len(modelPath) == 0 {
+		p.tree, p.instance2node = translate.GetTreeTestSet(false)
+	} else {
+		model, err := models.NewModelFromFile(modelPath)
+		if err != nil {
+			return nil, err
```

Comment:
nit: You unless properly wrapped by `models.NewModelFromFile`, you'll want to wrap this error with

```go
fmt.Errorf("could not read model file: %w", err)
```

If `models.NewModelFromFile(modelPath)` DOES wrap the error, I would just add a comment

```go
return nil, err // Wrapped by models.NewModelFromFile
```